docs/developer_README.md

```diff
@@ -0,0 +1,79 @@
+# LlamaBot: A Pythonic bot interface to LLMs
+
+This is a `developer` README for LlamaBot.
+
+LlamaBot implements a Pythonic interface to LLMs,
+making it much easier to experiment with LLMs in a Jupyter notebook
+and build Python apps that utilize LLMs.
+All models supported by [LiteLLM](https://github.com/BerriAI/litellm) are supported by LlamaBot.
+
+## Setting up your developer environment
+
+Make sure that you have followed the [README.md](../README.md) before jumping on the next steps.
+
+## Creating a Conda virtual environment
+
+It is recommended to have [Conda](https://conda.io/projects/conda/en/latest/user-guide/install/index.html) available on your system.
+
+```bash
+conda env create -f environment.yml
+```
+
+After the conda environment creation, you may need to run the following commands.
+
+```bash
+conda activate llamabot
+```
+
+The following command will install all your pip dependencies
+
+```bash
+pip install -e .
+```
+
+## Installing pre-commit hook
+
+It is recommended to install and use the [pre-commit](https://pre-commit.com/) hook if you plan to commit this project.
+This will check for any issues before committing your code. It is also one of the recommended developers' best practices.
+
+The following command will install pre-commit in your virtual environment.
+
+```bash
+pre-commit install
+```
+
+Now before you commit your changes, you may run the following command and resolve any issues that come up.
+
+```bash
+pre-commit run
+```
```

Comment:
Thanks for adding! There is one final change that I can see needing. Putting it below as a suggestion to make it easy for you to add it in:

```suggestion
Now before you commit your changes,
pre-commit will automatically run code checks for you.
If there are any issues that pre-commit finds,
such as missing line breaks or others,
please make the appropriate changes and re-commit the files.
```

Reply:
Updated.